R/loadFromMetaphlan.R

```diff
@@ -149,20 +149,27 @@ loadFromMetaphlan <- function(file, sample_meta = NULL, phy_tree = NULL, ...){
 }
 
 # Check that metaphlan file contains correct information
-.check_metaphlan <- function(data){
-    # Get rowdata columns
-    rowdata_columns <- data[ , 1:2]
-    # Get columns that go to assay
-    assay_columns <- data[ , 3:ncol(data)]
+.check_metaphlan <- function(x){
+    # Getting column indices of character columns
+    col_idx <- sapply(x, is.character)
+    # Get rowdata column indices
+    col_rowdata_idx <- col_idx[col_idx==TRUE]
+    # Getting column indices of numeric columns
+    col_idx <- sapply(x[,!col_idx], is.numeric)
+    # Get assay column indices
+    col_assay_idx <- col_idx[col_idx==TRUE]
     # Initialize result 
     result <- TRUE
     
-    # Check rowdata column names that they contain right information, and check that 
-    # rest of the columns represents abundances in samples.
+    # Check presence of other data than numeric and character
+    # Check rowdata exist
+    # Check assay abundance data exist
+    # Check rowdata column names that they contain right information.
     # If these requirements are met, give FALSE. Otherwise, give TRUE.
-    if( any(colnames(rowdata_columns) %in% "clade_name") && 
-        any(grepl("id", colnames(rowdata_columns))) && 
-        is.numeric(unlist(assay_columns)) ){
+    if( sum(col_idx==FALSE)==0 && 
```

Comment:
`sum(col_idx==FALSE)==0` -> how about just `!any(!col_idx)` or `all(col_idx)`

Reply:
I will switch this to all(col_idx) however this will depend on the previous step of `col_idx <- sapply(**x[,!col_idx]**, is.numeric)`
without it, it won't detect if there any other types than numeric and character.

Reply:
It is risky to convert all non-character variables into numerics here inside the function. 

What if you have factors, for instance? Like factor(c("A", "B", "C")). This would be converted into numeric c(1,2,3). Do we want to enforce such behavior?

Or should we catch and remove non-numeric fields already earlier, or provide a warning? 

I am now not sure why we want to consider other than character and numeric variables. What their purpose is in this context? Should we definitely allow them, or actively not allow them?

Reply:
Hi,
That's true, I didn't see that.
I agree, I hardly think metaphlan ouputs will contain other than numeric and character variables.
I reformulated the checks using `class` , I tested it for different set-ups and so far working, do you think this could be ok? (at latest commit)